app/src/main/java/com/omkarmoghe/pokemap/views/map/MapWrapperFragment.java

```diff
@@ -130,41 +127,52 @@ public void onLocationChanged(Location location) {
         locationFab.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                if (mLocation != null && mGoogleMap != null) {
+
+                Context context = getContext();
+
+                if (mLocation != null && mGoogleMap != null && context != null) {
                     mGoogleMap.animateCamera(CameraUpdateFactory.newLatLngZoom(
                             new LatLng(mLocation.getLatitude(), mLocation.getLongitude()), 15));
 
-                    MainActivity.toast.setText("Found you!");
-                    MainActivity.toast.show();
+                    Toast.makeText(context, "Found you!", Toast.LENGTH_SHORT).show();
                 }
                 else{
 
-                    MainActivity.toast.setText("Waiting on location...");
-                    MainActivity.toast.show();
+                    if (context != null) {
+                        Toast.makeText(context, "Waiting on location...", Toast.LENGTH_SHORT).show();
+                    }
                 }
             }
         });
 
-        mView.findViewById(R.id.closeSuggestions).setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                mView.findViewById(R.id.layoutSuggestions).setVisibility(View.GONE);
-            }
-        });
-
         return mView;
     }
+
+    private void showLocalizationFailed() {
```

Comment:
Well, GPS is also maybe not the best naming because some location information derives from mobile phone networks (triangulation) or wireless network etc. -- the code internally always calls the methods "ABCLocationXYZ" -- but I like the idea of changing "Locatization" wording and making it more precise what failed -- the "fetch" operation. 

I changed to, "*LocationFetchFailed()"